# Phototransistor

Herein we will explore the [phototransistor](https://cdn-shop.adafruit.com/product-files/2831/HW5P-1_2015__1_.pdf).  

The datasheet linked to above contains test results under the condition that $V_\mathrm{CC}=5\mathrm{V}$ and the resistance in series with the phototransistor is $R_\mathrm{s}=R=1000\Omega$.  One of those results is that the nominal current flowing through the device when exposed to $10\,\mathrm{Lux}$ is $50\mu\mathrm{A}$.  

Assuming we were to place a voltmeter with a resistance of $R_\mathrm{V}$ in parallel with a resistance of $R$ that is in series with the phototransistor, the total resistance in series with said device, denoted as '$R_\mathrm{s}$', is givne by:  

$$ G_\mathrm{s} =\frac{1}{R_\mathrm{s}}  =  \frac{1}{R} + \frac{1}{R_\mathrm{V}} = \frac{R + R_\mathrm{V}}{RR_\mathrm{V}} \quad\implies\quad R_\mathrm{s} = \frac{RR_\mathrm{V}}{R+R_\mathrm{V}}$$

Given that it is recommended that $\approx\! 1\mathrm{k}\Omega \le R_\mathrm{s} < 10\mathrm{k}\Omega$ and assuming $R_\mathrm{V}\approxeq 2\mathrm{k}\Omega$, what should $R$ be equal to?

$$ \approx\!1\mathrm{k}\Omega \le \frac{RR_\mathrm{V}}{R+R_\mathrm{V}} < 10\mathrm{k}\Omega$$

$$ \approx\!1\mathrm{k}\Omega\,(R+R_\mathrm{V}) \le RR_\mathrm{V} < 10\mathrm{k}\Omega\,(R+R_\mathrm{V})$$

Dividing throughout the above inequality by $R_\mathrm{V}\approxeq 2\mathrm{k}\Omega$ gives:  

$$ \approx\!0.5(R + 2\mathrm{k}\Omega) \le R < \approx\!5(R+2\mathrm{k}\Omega)$$

Subtracting throughout by $R$ gives:  

$$ \approx\!(1000\Omega - 0.5R)\le 0<\approx\!(4R + 10\mathrm{k}\Omega)$$

Finally,  

$$ R \ge \approx\!2000\Omega $$

Without assuming that $R_\mathrm{V}\approxeq 2000\Omega$,  

$$ \approx \!1\mathrm{k}\Omega\,\left(\frac{R}{R_\mathrm{V}} + 1\right) \le R < 10\mathrm{k}\Omega\,\left(\frac{R}{R_\mathrm{V}} + 1\right)$$

$$ \approx\!R\,\left(\frac{1\mathrm{k}\Omega}{R_\mathrm{V}} - 1\right) \le -1\mathrm{k}\Omega < R\,\left(\frac{10\mathrm{k}\Omega}{R_\mathrm{V}} - 1\right) + 9\mathrm{k}\Omega$$

$$ R\,\frac{R_\mathrm{V} - 1\mathrm{k}\Omega}{R_\mathrm{V}}\ge\approx 1\mathrm{k}\Omega\quad\quad\text{and}\quad\quad R\,\frac{R_\mathrm{V} - 10\mathrm{k}\Omega}{R_\mathrm{V}}<10\mathrm{k}\Omega$$

The [voltmeter](https://www.adafruit.com/product/3989) that we will be connecting in parallel to $R$ has a resistance of $R_\mathrm{V}\approxeq2.866\mathrm{k}\Omega$ according to our [multimeter](https://www.adafruit.com/product/308).

In [1]:
R_V = 2866
(R_V - 1000)/R_V

0.6510816468946267

In [2]:
(R_V-1e4)/R_V

-2.4891835310537336

In [3]:
1000*R_V/(R_V-1000)

1535.9056806002143

Thus,  

$$ R \ge\approx 1536\Omega  $$

Assuming $R\approxeq 10\mathrm{k}\Omega$,  

In [4]:
R = 50000
R_s = R*R_V/(R+R_V)
R_s

2710.626867930239

## Design  

Our circuit will consist of a $6\mathrm{VDC}$ power supply and a load consisting of a phototransistor in series with a $\approx 2.866\mathrm{k}\Omega$ voltmeter.

### Ancillaries  

A capacitor may be added in parallel to the load, which consists of the phototransistor in series with the resistance $R_\mathrm{s}$, and source in addition to another capacitor added in parallel to said resistance.  These add current stability, protecting both the phototransistor and the resistance from abrubt spikes in current caused by spikes in voltage.

With regards to the capacitance, denoted as '$C_\mathrm{s}$', in parallel with $R_\mathrm{s}$, according to the aforementioned datasheet, both the rise time and fall time, denoted as '$t_\mathrm{r}$' and '$t_\mathrm{f}$' respectively, are equal to $2\mu\mathrm{s}$, implying that a change of $\Delta U_\mathrm{s}$ occurs over $\Delta t = 2\mu\mathrm{s}$.  The average current drawn by $C_\mathrm{s}$ during this timeframe is thus given by:  

$$ i_\mathrm{avg} = C_\mathrm{s}\,\frac{\Delta U_\mathrm{s}}{\Delta t}$$

### Rationale

There is an impulse excitation of $x(t) = E\mathbf{1}(t)$, where $E=6\mathrm{V}$ and $\mathbf{1}(t)$ is the unit step function, caused by connecting the power supply to the load.  Assuming there is a practical capacitor with a capacitance of $C$ and a resistance of $R_C$ in parallel with the load and source, via KVL and element equations,  

$$ u = u(t) = i_\mathrm{V}R_\mathrm{V} + u_\mathrm{P} = R_Ci_C + \frac{1}{C}\,\int i_C\,\mathrm{d}t = E\mathbf{1}(t) - R_\mathrm{t}i$$

Here, '$u_\mathrm{P}$' denotes the voltage dropped across the phototransistor.

Via KCL,  

$$ i = i_C + i_\mathrm{V}$$

Laplace transforming throughout the above two equation gives:  

$$ U(s) =R_\mathrm{V} I_\mathrm{V}(s) + U_\mathrm{P}(s) = R_CI_C(s) + \frac{I_C(s)}{sC} + u_C(0) = \frac{E}{s} - R_\mathrm{t}I(s)$$

and,  

$$ I(s) = I_C(s) + I_\mathrm{V}(s)$$

Via the above two equations and substitution,  

$$ $$

## Test Results

The first test consisted of a phototransistor connected to a $V_\mathrm{CC}=6\mathrm{V}$ DC voltage source in series with a voltmeter with a resistance of $R_\mathrm{V}\approxeq 2.866\mathrm{k}\Omega$.  It might be worth noting that a $10\mu\mathrm{F}\;50\mathrm{V}$ capacitor was connected in parallel with the load and source for current stability.  It also might be a good idea to add another capacitor in parallel with the voltmeter/resistance that is in series with the phototransistor.

When exposed to ambient room light $\approx\!50\,\mathrm{lx}$, the voltmeter indication is $U_\mathrm{V}\approxeq 0.5\mathrm{V}$, implying a current of $I_\mathrm{V} = I = G_\mathrm{V}U_\mathrm{V} \approxeq 174.459\mu\mathrm{A}$.

In [5]:
0.5/2866

0.00017445917655268666

Now, suppose there were a $10\mu\mathrm{F}\;50\mathrm{V}$ capacitor in parallel with our voltmeter.  The average current drawn by said capacitor when going from $\approx\!0\mathrm{lx}$ to $\approx\!50\mathrm{lx}$ is given by:  

$$ i_\mathrm{avg} \approxeq 10\mu\mathrm{F}\,\frac{0.5\mathrm{V}}{2\mu\mathrm{s}} = 2.5\mathrm{A}$$

## Conclusion

As it turns out, adding a capacitor in parallel with the resistance is a bad idea!  It may also be worthwhile to examine the impact that the other capacitor has on the circuit.